multi_partner_learning.py

```diff
@@ -208,10 +220,35 @@ def compute_test_score(self):
         # Plot training history # TODO: move the data saving and plotting in dedicated functions
         if self.is_save_data:
             self.save_data()
+        
+        self.save_final_model_weights(model_to_evaluate)
 
         logger.info("Training and evaluation on multiple partners: done.")
         end = timer()
         self.learning_computation_time = end - start
+        
+    def save_final_model_weights(self, model_to_save):
+        """Save final model weights"""
+        #TODO: Have the last metrics in the model filename? 
+        
+        model_folder = os.path.join(self.save_folder, 'model')
+
+        if not os.path.isdir(model_folder):
+            os.makedirs(model_folder)
+            
+        model_to_save.save_weights(os.path.join(model_folder, self.dataset_name+'_final_weights.h5'))
+        model_weights = model_to_save.get_weights()
+        
+            
+        # model_weights = []
+        # for layer in model_to_save.layers:
+        #     layer_weights = []
+        #     for weight in layer.get_weights():
+        #         layer_weights.append(weight)
+        #     model_weights.append(layer_weights)
+            
+        np.save(os.path.join(model_folder, self.dataset_name+'_final_weights.npy'), 
+                model_weights)
```

Comment:
Why both `.h5` and `.npy` formats?

Reply:
I was thinking that to draw the layer weights it could be easier. But not sure. I can remove the .npy saving.